bioio_base/benchmark.py

```diff
@@ -89,62 +95,63 @@ def benchmark_test(
     }
 
 
-# Definitions of benchmark tests
-# ran for each test file
-BENCHMARK_DEFINITIONS: typing.List[BenchmarkDefinition] = [
-    {
-        "prefix": "First Scene Read",
-        "test": lambda file, reader: reader(file).get_image_data(),
-    },
-    {
-        "prefix": "All Scenes Read",
-        "test": _all_scenes_read,
-    },
-    {
-        "prefix": "First Scene Delayed Read",
-        "test": lambda file, reader: reader(file).get_image_dask_data(),
-    },
-    {
-        "prefix": "All Scenes Delayed Read",
-        "test": _all_scenes_delayed_read,
-    },
-    {
-        "prefix": "Metadata Read",
-        "test": lambda file, reader: reader(file).metadata,
-    },
-    {
-        "prefix": "OME Metadata Read",
-        "test": _read_ome_metadata,
-    },
-]
-
-
 def benchmark(
     reader: typing.Type[Reader],
-    test_file_dir: pathlib.Path,
+    test_files: typing.List[bioio_base.types.PathLike],
     additional_test_definitions: typing.List[BenchmarkDefinition] = [],
+    output_destination: str = OUTPUT_DESTINATION_DEFAULT,
 ) -> None:
     """Perform actual benchmark test"""
     benchmark_start_time = time.perf_counter()
 
     # Ensure test files are present
-    assert (
-        test_file_dir.exists()
-    ), f"Test resources directory can't be found: {test_file_dir}"
-    assert any(
-        test_file_dir.iterdir()
-    ), f"Test resources directory is empty: {test_file_dir}"
+    assert len(test_files) > 0, "Test file list is empty"
+
+    # Default benchmark test definitions
+    default_benchmark_tests: typing.List[BenchmarkDefinition] = [
```

Comment:
Moving these here made it so the "test" function doesn't need to be handed the reader which is handy since the plugins already know which reader they'd like to use